imap_processing/ultra/l1c/ultra_l1c.py

```diff
@@ -0,0 +1,50 @@
+"""Calculates ULTRA L1c."""
+
+from imap_processing.ultra.l1c.histogram import calculate_histogram
+from imap_processing.ultra.l1c.pset import calculate_pset
+
+
+def ultra_l1c(data_dict: dict):
+    """
+    Process ULTRA L1A and L1B data into L1C CDF files at output_filepath.
+
+    Parameters
+    ----------
+    data_dict : dict
+        The data itself and its dependent data.
+
+    Returns
+    -------
+    output_datasets : list of xarray.Dataset
+        List of xarray.Dataset
+    """
+    output_datasets = []
+    instrument_id = 45 if any("45" in key for key in data_dict.keys()) else 90
+
+    if (
+        f"imap_ultra_l1a_{instrument_id}sensor-histogram" in data_dict
+        and f"imap_ultra_l1b_{instrument_id}sensor-cullingmask" in data_dict
+    ):
+        histogram_dataset = calculate_histogram(
+            data_dict,
+            f"imap_ultra_l1c_{instrument_id}sensor-histogram",
+        )
+        output_datasets.append(histogram_dataset)
+    elif (
+        f"imap_ultra_l1b_{instrument_id}sensor-cullingmask" in data_dict
+        and f"imap_ultra_l1b_{instrument_id}sensor-de" in data_dict
+        and f"imap_ultra_l1b_{instrument_id}sensor-extspin" in data_dict
+    ):
+        pset_dataset = calculate_pset(
+            data_dict, f"imap_ultra_l1c_{instrument_id}sensor-pset"
+        )
+
+        output_datasets.extend(
+            [
+                pset_dataset,
+            ]
+        )
```

Comment:
You are doing append/extend on single iteration items. You can just set the object directly to the list you want and I think it will be more clear. You also use `append` above this, so if you don't want to set explicitly then I'd suggest using `append` here insted of `extend`
```suggestion
        output_datasets = [pset_dataset]
```